src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -18629,5 +18861,1539 @@ public static void Main()
+                context.RegisterSymbolAction(Handle, SymbolKind.Parameter);
+            }
+
+            private void Handle(SymbolAnalysisContext context)
+            {
+                switch (context.Symbol.ToTestDisplayString())
+                {
+                    case "A..ctor([System.Int32 X = 0])":
+                        Interlocked.Increment(ref FireCount1);
+                        break;
+                    case "System.Int32 A.X { get; init; }":
+                        Interlocked.Increment(ref FireCount2);
+                        break;
+                    case "[System.Int32 X = 0]":
+                        Interlocked.Increment(ref FireCount3);
+                        break;
+                    case "C..ctor([System.Int32 Z = 4])":
+                        Interlocked.Increment(ref FireCount4);
+                        break;
+                    case "[System.Int32 Z = 4]":
+                        Interlocked.Increment(ref FireCount5);
+                        break;
+                    default:
+                        Assert.True(false);
+                        break;
+                }
+            }
+        }
+
+        [Fact]
+        public void AnalyzerActions_03()
+        {
+            var text1 = @"
+record A(int X = 0)
+{}
+
+record C
+{
+    C(int Z = 4)
+    {}
+}
+";
+
+            var analyzer = new AnalyzerActions_03_Analyzer();
+            var comp = CreateCompilation(text1);
+            comp.GetAnalyzerDiagnostics(new[] { analyzer }, null).Verify();
+
+            Assert.Equal(1, analyzer.FireCount1);
+            Assert.Equal(1, analyzer.FireCount2);
+            Assert.Equal(0, analyzer.FireCount3);
+            Assert.Equal(1, analyzer.FireCount4);
+            Assert.Equal(0, analyzer.FireCount5);
+            Assert.Equal(1, analyzer.FireCount6);
+            Assert.Equal(1, analyzer.FireCount7);
+            Assert.Equal(1, analyzer.FireCount8);
+        }
+
+        private class AnalyzerActions_03_Analyzer : DiagnosticAnalyzer
+        {
+            public int FireCount1;
+            public int FireCount2;
+            public int FireCount3;
+            public int FireCount4;
+            public int FireCount5;
+            public int FireCount6;
+            public int FireCount7;
+            public int FireCount8;
+
+            private static readonly DiagnosticDescriptor Descriptor =
+               new DiagnosticDescriptor("XY0000", "Test", "Test", "Test", DiagnosticSeverity.Warning, true, "Test", "Test");
+
+            public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            => ImmutableArray.Create(Descriptor);
+
+            public override void Initialize(AnalysisContext context)
+            {
+                context.RegisterSymbolStartAction(Handle1, SymbolKind.Method);
+                context.RegisterSymbolStartAction(Handle1, SymbolKind.Property);
+                context.RegisterSymbolStartAction(Handle1, SymbolKind.Parameter);
```

Comment:
Can we add a `context.RegisterSymbolStartAction(..., SymbolKind.NamedType)` with a nested `RegisterSymbolEndAction` and then verify that all the `FireCountX` are `0` at the start of the former callback and `1` at the start of latter callback? This should verify correct ordering of callbacks.